renku_notebooks/api/notebooks.py

```diff
@@ -130,8 +132,52 @@ def launch_notebook(user):
         )
 
     # set the notebook image
-    image = get_notebook_image(user, namespace, project, commit_sha)
-
+    if requested_image is None:
+        # try to use image tied to the current commit from renku gitlab
+        check_image = get_notebook_image(gl_project, None, commit_sha[:7])
+        if check_image is None:
+            # the image tied to the current commit does not exist, send 404 response
+            return current_app.response_class(
+                status=404,
+                response="Cannot find the image tied to commit "
+                f"{commit_sha[:7]} for user {user['name']}.",
+            )
+        else:
+            # the image for the current commit exists, use it
+            image = check_image
+            is_image_private = gl_project.attributes.get("visibility") in {
+                "private",
+                "internal",
+            }
+    else:
+        # a specific image name has been passed, check if it exists
+        url, image_name, tag = parse_image_name(requested_image)
+        # an image name can be more than namespace/project, i.e. namespace/project/my/image
+        # but namespace/project is needed to get the gitlab project from the gitlab python sdk
+        image_name_split = image_name.split("/")
+        namespace_project = (
+            "/".join(image_name_split[:2]) if len(image_name_split) >= 2 else image_name
```

Comment:
I would avoid the reverse engineering of the project namespace and name all together when an image is provided. Instead I would just continue using the registry API (the one of the actual registry, not the one which is part of the GitLab API) and check if the user can access the image, first without her personal oauth token (if that works it's a public image), then with her personal access token supplied as `Authorization: bearer <oauth-token>` header in the request which queries the manifest. This last step would only be done if the token token endpoint returned in the `Www-Authenticate` header of the 401 response is part of Renku's GitLab instance. 
I think that would make the code a bit simpler, more easily extensible to multiple/other git providers and be a bit more consistent because it uses the exact same API / authorization mechanism for determining the users access to the image in the registry that is later on used by k8s to actually pull the image.

Reply:
With the latest changes I tested requesting a gitlab image with nested subgroups and it works.